src/app/shared/pipes/default-null-or-empty/default-null-or-empty.pipe.ts

```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({ name: 'defaultNullOrEmpty' })
+export class DefaultNullOrEmpty implements PipeTransform {
+  transform(value: any): any {
+    if (typeof value === 'string') {
+      return value.trim().length > 0 ? value : '-';
```

Comment:
Should the positive return be `value.trim()`?

Reply:
As this pipe is for formatting / display, we don't want to return `value.trim()` in this situation because the underlying data could still have leading or trailing spaces